blackbox/handlers/storage/dropbox.py

```diff
@@ -26,9 +27,25 @@ def __init__(self, **kwargs):
 
         self.upload_base = self.config.get("upload_directory") or "/"
         self.client = DropboxClient(self.config["access_token"])
+        self.valid = self._validate_token()
+
+    def _validate_token(self):
+        """Check if dropbox token is valid."""
+        try:
+            return self.client.check_user("test").result == "test"
+        except AuthError:
+            return False
 
     def sync(self, file_path: Path) -> None:
         """Sync a file to Dropbox."""
+        # Check if Dropbox token is valid.
+        if self.valid is False:
+            error = "Dropbox token is invalid"
+            self.success = False
+            self.output = error
+            log.warning(error)
```

Comment:
I don't think warning is the correct log level. 
- A warning is for informing users that something unexpected happened, but everything should still be working more or less as normal.
- An error is for informing the user that something is definitely wrong, but it will not prevent the entire application from running.. probably.
- Critical is for errors that prevent the entire application from running.

Reply:
Thank you, will use the correct level from now on.